pkg/utils/ingress.go

```diff
@@ -15,113 +15,33 @@
 package utils
 
 import (
-	"os"
+	"context"
 
-	extension "k8s.io/api/extensions/v1beta1"
-	networking "k8s.io/api/networking/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/kubernetes"
 )
 
-var IngressApiMap = map[string]string{
-	"corev1":      CoreV1IngressInformer,
-	"extensionv1": ExtV1IngressInformer,
-}
-
-var (
-	ExtensionsIngress = schema.GroupVersionResource{
-		Group:    "extensions",
-		Version:  "v1beta1",
-		Resource: "ingresses",
-	}
-
-	NetworkingIngress = schema.GroupVersionResource{
-		Group:    "networking.k8s.io",
-		Version:  "v1beta1",
-		Resource: "ingresses",
-	}
-)
-
-func GetIngressApi(kc kubernetes.Interface) string {
-	ingressAPI := os.Getenv("INGRESS_API")
-	if ingressAPI != "" {
-		ingressAPI, ok := IngressApiMap[ingressAPI]
-		if !ok {
-			return CoreV1IngressInformer
-		}
-		return ingressAPI
-	}
-
-	var timeout int64
-	timeout = 120
-
-	_, ingErr := kc.NetworkingV1beta1().Ingresses("").List(metav1.ListOptions{TimeoutSeconds: &timeout})
-	if ingErr != nil {
-		AviLog.Infof("networkingv1 ingresses not found, setting informer for extensionsv1: %v", ingErr)
-		return ExtV1IngressInformer
-	}
-	return CoreV1IngressInformer
-}
-
-func fromExtensions(old *extension.Ingress) (*networking.Ingress, error) {
-	networkingIngress := &networking.Ingress{}
-
-	err := runtimeScheme.Convert(old, networkingIngress, nil)
-	if err != nil {
-		return nil, err
-	}
-
-	return networkingIngress, nil
-}
-
-func fromNetworking(old *networking.Ingress) (*extension.Ingress, error) {
-	extensionsIngress := &extension.Ingress{}
-
-	err := runtimeScheme.Convert(old, extensionsIngress, nil)
-	if err != nil {
-		return nil, err
-	}
-
-	return extensionsIngress, nil
-}
-
-// ToNetworkingIngress converts obj interface to networking.Ingress
-func ToNetworkingIngress(obj interface{}) (*networking.Ingress, bool) {
-	oldVersion, inExtension := obj.(*extension.Ingress)
-	if inExtension {
-		ing, err := fromExtensions(oldVersion)
-		if err != nil {
-			AviLog.Warnf("unexpected error converting Ingress from extensions package: %v", err)
-			return nil, false
-		}
-
-		return ing, true
-	}
+var ingressClassEnabled *bool
 
-	if ing, ok := obj.(*networking.Ingress); ok {
-		return ing, true
+func IsIngressClassEnabled(kc ...kubernetes.Interface) bool {
```

Comment:
This is used both for setting ingressClassEnabled  and checking if ingressClass is Enabled. 
Not very important, but may be we can improve this a bit. I got confused at first. Can we have two different functions one for setting and another for getting the ingress class ?